prover/src/trace/mod.rs

```diff
@@ -51,17 +54,6 @@ pub trait Trace: Sized {
     /// Returns a reference to a [Matrix] describing the main segment of this trace.
     fn main_segment(&self) -> &ColMatrix<Self::BaseField>;
 
-    /// Builds and returns the auxiliary trace segment. If the trace does not require an auxiliary
-    /// segment, None is returned.
-    ///
-    /// The `rand_elements` slice contains the random elements to use to build the aux segment. If a
-    /// Lagrange kernel column is present, the `lagrange_kernel_rand_elements` should be used.
-    fn build_aux_segment<E: FieldElement<BaseField = Self::BaseField>>(
-        &mut self,
-        rand_elements: &[E],
-        lagrange_kernel_rand_elements: Option<&[E]>,
-    ) -> Option<ColMatrix<E>>;
```

Comment:
What is the benefit of replacing this function with `AuxTraceBuilder` trait?

Reply:
The main reason was as to not pollute the `Trace` trait, since now building the auxiliary trace requires 3 associated types: `AuxRandElements<E>`, `AuxParams`, and `AuxProof` that are only used with `build_aux_trace()`.

We could put it back in `Trace`, but then building a simple trace would require a lot of unnecessary dummy types:

```rs
impl Trace for SimpleTrace {
  type AuxRandElements<E> = PhantomData<E>;
  type AuxParams = ();
  type AuxProof = ();

  // ...
}
```

which could be okay too?